src/Common/tests/System/IO/Compression/CompressionStreamPerfTestBase.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.IO.Compression
+{
+    public abstract class CompressionStreamPerfTestBase : CompressionStreamTestBase
+    {
+        public static IEnumerable<object[]> CanterburyCorpus()
+        {
+            foreach (CompressionLevel compressionLevel in Enum.GetValues(typeof(CompressionLevel)))
+            {
+                foreach (int innerIterations in new int[] { 1, 10 })
+                {
+                    foreach (var fileName in UncompressedTestFiles())
+                    {
+                        yield return new object[] { innerIterations, fileName[0], compressionLevel };
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Benchmark tests to measure the performance of individually compressing each file in the
+        /// Canterbury Corpus
+        /// </summary>
+        [Benchmark]
+        [MemberData(nameof(CanterburyCorpus))]
+        public void Compress_Canterbury(int innerIterations, string uncompressedFileName, CompressionLevel compressLevel)
+        {
+            byte[] bytes = File.ReadAllBytes(uncompressedFileName);
+            MemoryStream[] compressedDataStreams = new MemoryStream[innerIterations];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                // resizing a memory stream during compression will throw off our results, so pre-size it to the
+                // size of our input
+                for (int i = 0; i < innerIterations; i++)
+                    compressedDataStreams[i] = new MemoryStream(bytes.Length);
```

Comment:
bytes.Length might be too in some cases little to avoid reallocations

Reply:
The only time this will trigger a resize is when the compressed data is larger than its corresponding uncompressed data. That's not a case that we hit in our Canterbury data set, so it isn't a worry.